MyHashMap.py

```diff
@@ -1,38 +1,66 @@
 class MyHashMap:
-    # Initialize your data structure here
-    def __init__(self):
-        self.keys = []
-        self.values = []
+	# Initialize your data structure here
+	# Here we can even fix the length of the horizontal linked list to 8 and then do the same thing
+	# then we have to implement rehashing and everything
+	class Node(object):
+		def __init__(self, key, val):
+			self.key = key
+			self.val = val
+			self.next = None
 
-    # value will always be non-negative
-    def put(self, key, value):
-        if key in self.keys:
-            self.values[self.keys.index(key)] = value
-        else:
-            self.keys.append(key)
-            self.values.append(value)
+	def __init__(self):
+		self.Hashlist = [None] * 10000
+	
+	# value will always be non-negative
+	def put(self, key, value):
+		i = self.find_index(key)
+		if self.Hashlist[i] == None:
+			self.Hashlist[i] = self.Node(-1, -1)
+		prev_node = self.find_node(self.Hashlist[i], key)
+		if prev_node.next == None:
+			prev_node.next = self.Node(key, value)
+		else:
+			prev_node.next.val = value
 
-    # Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key
-    def get(self, key):
-        if key in self.keys:
-            return self.values[self.keys.index(key)]
-        else:
-            return -1
+	# Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key
+	def get(self, key):
+		i = self.find_index(key)
+		if self.Hashlist[i] == None:
+			return -1
+		prev_node = self.find_node(self.Hashlist[i], key)
+		# why should we check for this condition again because node.next would never be none if self.Hashlist[i] is not none
+		# And the length of our horizontal list is either zero or minimum two nodes right
```

Comment:
we need to check this condition because :
lets see what we are actually doing from the pur function 
we are first finding the index for the key we need to insert lets say the key with us is 2 now we went to the index 2 of our array if we find any other element already over there we are moving furthur in our nodes and inserting it at a place where there was null before
so now when i searched for that element i first checked the place where it was inserted if we find a null there that clearly means that key doesnt exist in our hashmap else it would be there
but if its not null over there i checked for nodes connected to it and traversed till i either got the node or i got null
in case i got null it means that key is not in our array so we again should check for this prev_node.next==null condition
and if its not null it means we have got the key and so we will return the value